controllers/admin/AdminAdyenOfficialPrestashopLogFetcherController.php

```diff
@@ -0,0 +1,286 @@
+use PrestaShop\PrestaShop\Adapter\CoreException;
+use PrestaShopBundle\Utils\ZipManager;
+
+require_once _PS_ROOT_DIR_ . '/modules/adyenofficial/vendor/autoload.php';
+
+class AdminAdyenOfficialPrestashopLogFetcherController extends ModuleAdminController
+{
+    /** @var \Adyen\PrestaShop\service\adapter\classes\Configuration $configuration */
+    private $configuration;
+
+    /** @var Adyen\PrestaShop\infra\Crypto */
+    private $crypto;
+
+    /** @var VersionChecker */
+    private $versionChecker;
+
+    /** @var string $logsDirectory */
+    private $logsDirectory;
+
+    /**
+     * AdminAdyenPrestashopCronController constructor.
+     *
+     * @throws CoreException|PrestaShopException
+     */
+    public function __construct()
+    {
+        $this->configuration = ServiceLocator::get('Adyen\PrestaShop\service\adapter\classes\Configuration');
+        $this->crypto = ServiceLocator::get('Adyen\PrestaShop\infra\Crypto');
+        $this->versionChecker = ServiceLocator::get('Adyen\PrestaShop\application\VersionChecker');
+        if ($this->versionChecker->isPrestaShop16()) {
+            $this->logsDirectory = _PS_ROOT_DIR_ . '/log/adyen';
+        } else {
+            $this->logsDirectory = _PS_ROOT_DIR_ . '/var/logs/adyen';
+        }
+
+        // Required to automatically call the renderView function
+        $this->display = 'view';
+        $this->bootstrap = true;
+        $this->toolbar_title[] = 'Adyen Logs';
+        parent::__construct();
+
+        // Zip download is triggered when queryParameter contains the download string
+        if ((string)Tools::getValue('download')) {
+            $this->createCurrentApplicationInfoFile();
+            $this->zipAndDownload(Tools::getValue('include-all'));
+            exit;
+        }
+    }
+
+    /**
+     * Render the log-fetcher template
+     *
+     * @return false|string
+     * @throws SmartyException
+     */
+    public function renderView()
+    {
+        $smartyVariables = array(
+            'logo' => Media::getMediaPath(_PS_MODULE_DIR_ . $this->module->name . '/views/img/adyen.png'),
+            'downloadUrl' => $this->getDownloadUrl()
+        );
+        $this->addCSS('modules/' . $this->module->name . '/views/css/adyen_admin.css');
+
+        // Passing variables in this call (instead of assign()) required for 1.6
+        $tpl = $this->context->smarty->createTemplate(
+            _PS_MODULE_DIR_ . $this->module->name . '/views/templates/admin/log-fetcher.tpl',
+            null,
+            null,
+            $smartyVariables
+        );
+
+        return $tpl->fetch();
+    }
+
+    /**
+     * Zip all files in the adyen log directory and download. If includeAll is set, zip all logs
+     *
+     * @param $includeAll
+     */
+    private function zipAndDownload($includeAll)
+    {
+        if ($includeAll) {
+            $this->logsDirectory = substr($this->logsDirectory, 0, strrpos($this->logsDirectory, '/'));
```

Comment:
would it make sense to have the base log folder like: `$this->logsDirectory = _PS_ROOT_DIR_ . '/log';` and then add the adyen instead of going up with 1 folder when it's include all?

Reply:
https://github.com/Adyen/adyen-prestashop/pull/185/commits/80244b3b44dd27a96524dd4077bf49e8d72d67e7